src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -576,14 +576,17 @@ public ProjectState WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyz
                 projectInfo: this.ProjectInfo.WithAnalyzerReferences(analyzerReferences).WithVersion(this.Version.GetNewerVersion()));
         }
 
-        public ProjectState AddDocument(DocumentState document)
+        public ProjectState AddDocuments(ImmutableArray<DocumentState> documents)
```

Comment:
Shouldn't we keep `AddDocument` to avoid allocation a single element array and bunch of enumerables when adding just one document?